File: lab_02/src/main.py
import time

def compute(x):
    return x*x - x*x + 4*x - 5*x + x + x

def main():
    while True:
        user_input = input("Enter number of iterations (integer): ")
        n = int(user_input)

        start = time.perf_counter()
        result = 0
        for i in range(n):
            result = compute(i)   
        end = time.perf_counter()

        elapsed_us = (end - start) * 1_000_000
        print(f"Result: {result}")
        print(f"N= {n}, Time: {elapsed_us:.0f} mcs")

        choice = input("Run again? (y/n): ").lower()
        if choice != 'y':
            break

if __name__ == "__main__":
    main()

File: lab_02/src/main.cpp
#include <iostream>
#include <chrono>
#include <limits>
#include <cmath>

using namespace std;
using namespace chrono;

double compute(double x){
    return x*x - x*x + x*4 - x*5 + x + x;
}

int main() {
    bool again = true;
    while (again) {
        cout << "N= ";
        double userInput;
        cin >> userInput;

        int n = userInput;
        auto start = high_resolution_clock::now();

        double result = 0.0;
        for (int i = 0; i < n; ++i) {
            result = compute(static_cast<double>(i));
        }

        auto end = high_resolution_clock::now();
        auto duration = duration_cast<microseconds>(end - start);

        cout << "Result: " << result << endl;
        cout << "N= " << n << " Time all " << duration.count() << " mcs" << endl;

        cout << "Run again? (y/n): ";
        char choice;
        cin >> choice;
        if (choice != 'y' && choice != 'Y') {
            again = false;
        }
        cin.ignore(numeric_limits<streamsize>::max(), '\n');
    }
    return 0;
}
